### DISCUSSION TITLE
How to exclude/ignore vendored code from cue eval

### DISCUSSION THREAD

**Discussion by solsson** [cue-lang/cue#1749] [Q&A] [answered]
With a [package](https://cuelang.org/docs/concepts/packages/#packages), or `cue eval ./...`, can I somehow prevent vendored code from being included? For example with Node.js something installed to `node_modules` could have a .cue file with the same `package` name that I use in my module.

**Comment by myitcv** [comment, marked as answer]
Just to be clear, you are not talking about the code vendored in `cue.mod/pkg/...`, but rather other directories contained by the module like `node_modules` (which is "vendored" in some cases), correct?

If so, the immediate fix is to create a `cue.mod` directory within the directory you want to ignore (which also has the effect of eliminating all subdirectories of that directory).

I will raise a feature request to explore a more principled solution (not that the `cue.mod` solution is wrong per se, but with CUE we have slightly more flexibility in that `cue.mod` is a directory)

cc @mvdan given discussion about this in the Go world.

**Reply by myitcv** [reply]
Actually, I'm going to avoid creating a separate issue for now, because I think the `cue.mod` approach works just fine. I've read through the various Go threads on this topic, and there doesn't seem a compelling reason why using the `cue.mod` approach (obviously `go.mod` in the case of Go) doesn't work.

**Comment by solsson** [comment]
> Just to be clear, you are not talking about the code vendored in cue.mod/pkg/..., but rather other directories contained by the module like node_modules (which is "vendored" in some cases), correct?

Yes, thank you for clarifying.

> If so, the immediate fix is to create a cue.mod directory within the directory you want to ignore (which also has the effect of eliminating all subdirectories of that directory).

That's actually sufficient granularity for the monorepo structures I have in mind.

> I will raise a feature request to explore a more principled solution (not that the cue.mod solution is wrong per se, but with CUE we have slightly more flexibility in that cue.mod is a directory)

Yes that's interesting. Includes/excludes with git semantics is quite useful but given cue.mod it seems reasonable to avoid having dotfiles at arbitrary locations.
